### DISCUSSION TITLE
serde_as with the CSV crate?

### DISCUSSION THREAD

**Discussion by BuyMyMojo** [jonasbb/serde_with#501] [Q&A] [answered]
So I have a custom FormInto implementation to handle a csv file but when decerialising it completely ignores my serde_as macro, what can I do to fix this? I just get a `invalid float literal` error before it even tries to use serde_as

implementation:
```rs
#[derive(Deserialize, Serialize, Debug)]
struct OptionalFloat(pub String);

impl From<OptionalFloat> for Option<f64> {
    fn from(val: OptionalFloat) -> Self {
        println!("{:?}", val); // This does not fire at all

        match val.0.trim() {
            "Error" | "NA" => None,
            s => Some(s.parse::<f64>().unwrap()),
        }
    }
}
```

section of my struct failing:
```rs
#[serde_as(as = "FromInto<OptionalFloat>")]
#[serde(rename = "GPU1Clk(MHz)")]
pub gpu1clk_mhz: Option<f64>,
```

**Comment by jonasbb** [comment]
Could you please post a small example which exhibits the problem? Otherwise, I cannot reproduce the issue. Please include a CSV with some failing lines and the struct which contains the `gpu1clk_mhz` field.

**Comment by BuyMyMojo** [comment, marked as answer]
I'm making a tool to do stuff with FrameViewCSV files
[FrameView_WindowsTerminal.exe_2022_08_13T002652_Log.csv](https://github.com/jonasbb/serde_with/files/9331593/FrameView_WindowsTerminal.exe_2022_08_13T002652_Log.csv)

Full file using the macro: https://github.com/BuyMyMojo/frame_view_tools/blob/d9f50b6ea60359c5bba42dea48fc4563a342661f/src/main.rs


I have got it working now but not with the macro:
https://github.com/BuyMyMojo/frame_view_tools/blob/main/src/main.rs

**Reply by jonasbb** [reply]
The order of these two lines is wrong. `#[serde_with::serde_as]` always needs to be *before* the derive.
```rust
#[derive(Default, Debug, Clone, PartialEq, Serialize, Deserialize)]
#[serde_as]
```

The linked files are too big for me to really help you. A more minimal example is necessary.
